usecase/create_todo_usecase.go

```diff
@@ -0,0 +1,40 @@
+package usecase
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"time"
+	"todo-clean/domain"
+	"todo-clean/lib/error_lib"
+)
+
+// TODO Use Func Wrapper Error!
+func (n *newUseCase) CreateTodoUseCase(ctx context.Context, todo domain.CreateTodoEntityRequest) (result *domain.CreateTodoEntity, err error) {
+
+	dbTx, err := n.repo.Begin(ctx)
+	if err != nil {
+
+		return nil, error_lib.WrapError("create todo usecase", err)
+	}
+	result, err = n.repo.CreateTodoRepository(dbTx, domain.CreateTodoEntity{
+		ID:          uuid.New(),
+		Title:       todo.Title,
+		Description: todo.Description,
+		Status:      todo.Status,
+		CreatedAt:   time.Now(),
+	})
+
+	if err != nil {
+		return nil, error_lib.WrapError("create todo usecase", err)
+	}
+	dbTx, err = n.repo.Commit()
+	if err != nil {
+		if rollbackErr := n.repo.RollBack(dbTx); rollbackErr != nil {
+
+			return nil, error_lib.WrapError("rollback create todo usecase", err)
```

Comment:
This should be const and rollback should wrap error then you don’t need to insert word like “rollback” because it will duplicate